base_layer/key_manager/src/mnemonic.rs

```diff
@@ -112,7 +118,8 @@ impl MnemonicLanguage {
                     // yielded from the initial word for this iteration
                     for language in languages {
                         let mut consistent = true;
-                        for compare in words {
+                        for compare_ind in 0..words.len() {
```

Comment:
Is it worth it to have `SeedWords` implement an iterator that provides word references? Then you don't need to mess around with indexes.

Reply:
See comment above.